udata/core/dataset/models.py

```diff
@@ -299,6 +299,8 @@ def clean(self):
         super(ResourceMixin, self).clean()
         if not self.urlhash or 'url' in self._get_changed_fields():
             self.urlhash = hash_url(self.url)
+        if not bool('name' in self.schema) ^ bool('url' in self.schema):
```

Comment:
We're not saying anything about version. Having a version and an url doesn't make sense. Not sure we want to make it any more complex though